From 116ce33a4eb0af5a67d173f97b40684b13d55c63 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Thu, 27 Aug 2020 14:28:42 +0800
Subject: [PATCH] Revert "drm: support atomic update flush for connector"

This reverts commit 511154506877c18f211a5f14bc6d95db40a1c73a.

For compatibility with GKI, connector atomic_begin/atomic_flush
should be removed.

Change-Id: I00b66338fd8a79230a71419ed476bf80e8496e2b
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/gpu/drm/drm_atomic_helper.c      | 24 ------------------------
 include/drm/drm_modeset_helper_vtables.h |  8 --------
 2 files changed, 32 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index cf8af4dcfae0..da0b303e2d28 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -2355,8 +2355,6 @@ void drm_atomic_helper_commit_planes(struct drm_device *dev,
 	struct drm_crtc_state *old_crtc_state, *new_crtc_state;
 	struct drm_plane *plane;
 	struct drm_plane_state *old_plane_state, *new_plane_state;
-	struct drm_connector *connector;
-	struct drm_connector_state *old_connector_state, *new_connector_state;
 	int i;
 	bool active_only = flags & DRM_PLANE_COMMIT_ACTIVE_ONLY;
 	bool no_disable = flags & DRM_PLANE_COMMIT_NO_DISABLE_AFTER_MODESET;
@@ -2432,28 +2430,6 @@ void drm_atomic_helper_commit_planes(struct drm_device *dev,
 
 		funcs->atomic_flush(crtc, old_crtc_state);
 	}
-
-	for_each_oldnew_connector_in_state(old_state, connector,
-					   old_connector_state,
-					   new_connector_state, i) {
-		const struct drm_connector_helper_funcs *funcs;
-
-		if (!connector->state->crtc)
-			continue;
-
-		if (!connector->state->crtc->state->active)
-			continue;
-
-		funcs = connector->helper_private;
-
-		if (!funcs || !funcs->atomic_flush)
-			continue;
-
-		DRM_DEBUG_ATOMIC("flushing [CONNECTOR:%d:%s]\n",
-				 connector->base.id, connector->name);
-
-		funcs->atomic_flush(connector, old_connector_state);
-	}
 }
 EXPORT_SYMBOL(drm_atomic_helper_commit_planes);
 
diff --git a/include/drm/drm_modeset_helper_vtables.h b/include/drm/drm_modeset_helper_vtables.h
index a772dea3afbf..7aea03bdbc22 100644
--- a/include/drm/drm_modeset_helper_vtables.h
+++ b/include/drm/drm_modeset_helper_vtables.h
@@ -1000,14 +1000,6 @@ struct drm_connector_helper_funcs {
 	 */
 	void (*atomic_commit)(struct drm_connector *connector,
 			      struct drm_connector_state *state);
-
-	/**
-	 * @atomic_flush:
-	 *
-	 * flush atomic update.
-	 */
-	void (*atomic_flush)(struct drm_connector *connector,
-			     struct drm_connector_state *conn_state);
 };
 
 /**
-- 
2.35.3

